app/Http/Controllers/Auth/AdmissionController.php

```diff
@@ -165,54 +168,82 @@ public function updateNote(Request $request, Application $application): Redirect
+                ->withoutGlobalScope('verified')
+                ->with('application')
+                ->whereIn('id', $not_admitted->pluck('user_id'))
+                //ignore users with any existing role
+                ->whereDoesntHave('roles')
+                ->orderBy('name')
+                ->get()
+        ]);
+    }
+
+
     /**
      * Accept and delete applciations.
      * @return RedirectResponse
      * @throws AuthorizationException
      */
     public function finalize(): RedirectResponse
     {
-        //        $this->authorize('finalizeApplicationProcess', User::class);
-        //        Cache::forget('collegists');
-        //        $not_handled_applicants = User::query()->withoutGlobalScope('verified')
-        //            ->where('verified', 0)
-        //            ->whereHas('application', function ($query) {
-        //                $query->where('submitted', true);
-        //            })
-        //            ->count();
-        //        if ($not_handled_applicants > 0) {
-        //            return redirect()->back()->with('error', 'Még vannak feldolgozatlan jelentkezések!');
-        //        }
-        //        DB::transaction(function () {
-        //            User::query()->withoutGlobalScope('verified')
-        //                ->where('verified', 0)
-        //                ->whereHas('application', function ($query) {
-        //                    $query->where('status', Application::STATUS_ACCEPTED);
-        //                })
-        //                ->update(['verified' => true]);
-        //            $usersToDelete = User::query()->withoutGlobalScope('verified')
-        //                ->where('verified', 0)->whereHas('application');
-        //            foreach ($usersToDelete->get() as $user) {
-        //                if ($user->profilePicture!=null) {
-        //                    Storage::delete($user->profilePicture->path);
-        //                    $user->profilePicture()->delete();
-        //                }
-        //            }
-        //            $files = File::where('application_id', '!=', null);
-        //            foreach ($files->get() as $file) {
-        //                Storage::delete($file->path);
-        //            }
-        //            $files->delete();
-        //            Application::query()->delete();
-        //            $usersToDelete->forceDelete();
-        //
-        //            RoleUser::where('role_id', Role::get(Role::APPLICATION_COMMITTEE_MEMBER)->id)->delete();
-        //            RoleUser::where('role_id', Role::get(Role::AGGREGATED_APPLICATION_COMMITTEE_MEMBER)->id)->delete();
-        //        });
-        //
-        //        Cache::clear();
-        // return back()->with('message', 'Sikeresen jóváhagyta az elfogadott jelentkezőket');
-        return back()->with('error', 'Még nincs implementálva.');
+        $this->authorize('finalize', Application::class);
```

Comment:
Here, we should check whether we have reached `$applicationDeadline->copy()->addWeek()`.

Or should we abolish this requirement? (I think it is useful, but I do not insist on it.) If not more, I think we should not allow finalisation while the application period is open. (The finalisation page could be shown, that is useful; but without the option to actually finalise applications.)

Reply:
I wanted to delete that part (maybe I left some parts of this logic). Originally, i just wanted to avoid accidental finalisations but now we have a separate page only for that, I think it is clear now.

Reply:
FWIW, removing that sounds okay to me (the one-week delay struck me as arbitrary when I saw it first)

Reply:
I think it would still be useful to prevent finalisation while the application period is open. But I can live without that; just remove the remainders and warnings related to that logic then.

Reply:
I have put it back but it only checks for the application deadline, does not add a week.

Reply:
Nice:)